src/browser/chrome.ts

```diff
@@ -40,16 +40,49 @@ export const closeBrowser = async () => {
 };
 
 const WINDOW_VARIABLE_NAME = `___webrender`;
+const LAST_RESPONSE_AT_VARIABLE_NAME = `__webrenderLastResponseAt`;
+const COUNTER_REQUEST_UNFINISHED_VARIABLE_NAME = `__webrenderCounterReqUnfinished`;
 export const getPageInitScriptFor = (
   js: string,
-  jsOn: JsOn = "commit"
-) => `try {
-  const AsyncFunction = Object.getPrototypeOf(
+  jsOn: JsOn = "commit",
+  timeout: number
+) => {
+  const PAGE_TRUE_LOAD = `
+  async pageTrueLoad ({
+    idleTimeout = 2000, resolveBeforeRenderingTimeout = 2000
+  } = {}) {
+    const timeoutPromise = new Promise((resolve) => {
+      setTimeout(() => {
+        resolve();
+      }, ${timeout} - resolveBeforeRenderingTimeout);
+    });
+
+    const loadPromise = new Promise((resolve) => {
+      addEventListener('load', () => {
+        const checkLoaded = setInterval(() => {
+          const lastResponseAt = +localStorage.getItem('${LAST_RESPONSE_AT_VARIABLE_NAME}');
+          const counterRequestsUnfinished = +localStorage.getItem('${COUNTER_REQUEST_UNFINISHED_VARIABLE_NAME}');
+          const isLoaded = lastResponseAt && Date.now() - lastResponseAt > idleTimeout;
```

Comment:
`isLoaded` name is confusing because it doesn't account for `counterRequestsUnfinished`. I.e. you send 4 requests, one of them finishes and 3 are pending for more than 2 seconds, then isLoaded becomes true.